### PR TITLE
Added command to create external beta tester groups

### PR DESCRIPTION
**Pull request by JeffreyGreenII**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
This change allows Pilot to ability to add external beta tester groups. Since iTunes no longer comes with default external tester groups, it's important to maintain an ability to add these groups manually to allow developers to setup their apps and those apps tests without accessing ITC's UI. This also opens up the possibility to have dynamic groups of external testers for specific builds and specific users.
I typed out the edge cases that would cause a problem (no input, missing input, invalid input, incorrect input) and test for all cases I suspected could happen.

### Description
I created a new manager class in Pilot called GroupManager with the single responsibility of handling groups, as TesterManager and BuildManager didn't feel apt for such a role. I added a command to the commands_generator in Pilot called "group". I also updated Spaceship's client to have the ability to access TestFlight's Groups API endpoint to POST a group name to a specific application. I accessed this new client function by adding static functionality to the Spaceship::TestFlight::Group class to handle a new group.

### CODE REVIEW

**Comment by JeffreyGreenII** [comment]
I signed it!

**Comment by JeffreyGreenII** [comment]
I signed it!

**Comment by JeffreyGreenII** [comment]
I signed it!

File: pilot/lib/pilot/options.rb [hunk 1]
```diff
@@ -156,7 +156,11 @@ def self.available_options
                                      type: Array,
                                      verify_block: proc do |value|
                                        UI.user_error!("Could not evaluate array from '#{value}'") unless value.kind_of?(Array)
-                                     end)
+                                     end),
+        FastlaneCore::ConfigItem.new(key: :group_name,
+                                     env_name: "PILOT_GROUP_NAME",
+                                     description: "The name of the group being made for external testers",
```
Comment (ohayon) [inline pilot/lib/pilot/options.rb]: Could we maybe make a more specific name and description for this option? Just dont want people to confuse it with the `:groups` option above.

**Review by ohayon — CHANGES_REQUESTED** [review]
Hey @JeffreyGreenII ! Thanks for this PR this would definitely be great to have in fastlane! Just one comment above, would be curious to hear your thoughts!

**Comment by ohayon** [comment]
Hey @JeffreyGreenII, I am going to close this PR for now as we have made a bunch of changes to how pilot works because of some more changes that Apple made to TestFlight and I'm not sure that this PR is necessary anymore. I definitely could be mistaken here, so please feel free to comment here and reopen it if necessary! Thanks! 🚀
